dhall/src/Dhall/Pretty/Internal.hs

```diff
@@ -1072,21 +1072,23 @@ prettyCharacterSet characterSet expression =
         angles . map prettyAlternative . Dhall.Map.toList
 
     prettyChunks :: Pretty a => Chunks Src a -> Doc Ann
-    prettyChunks (Chunks a b) =
-        if anyText (== '\n')
-        then
-            if anyText (/= '\n')
+    prettyChunks chunks@(Chunks a b)
+        | anyText (== '\n') =
+            if not (null a) || anyText (/= '\n')
```

Comment:
My understanding is that if `anyText (== '\n')` is `True` then `not (null a)` always evaluates to `True`, therefore the second branch of this `if` expression becomes dead code

Reply:
Not quite: For example in the case of `Chunks [] "\n"`, `anyText (== '\n')` is `True`, but `not (null a)` is `False` (a is `[]`).

This additional condition was necessary to fix idempotence:

    Formatting should be idempotent: FAIL (6.63s)
      *** Failed! Falsifiable (after 1534 tests and 2 shrinks):
      Unicode
      Header ""
      TextLit (Chunks [] "\n ")
      "''\n\n${\" \"}''\n" /= "\"\\n${\" \"}\"\n"
      Use --quickcheck-replay=854885 to reproduce.

Should I add a comment?